statix.solver/src/main/java/mb/statix/spec/ApplyRelaxed.java

```diff
@@ -50,9 +50,9 @@ class ApplyRelaxed extends ApplyMode<VoidException> {
 
         final IConstraint appliedBody;
         if(safety.equals(Safety.UNSAFE)) {
-            appliedBody = rule.body().unsafeApply(matchResult.substitution()).withCause(cause);
+            appliedBody = rule.body().unsafeApply(matchResult.substitution(), true).withCause(cause);
```

Comment:
I am wondering if there are cases where either mode of application is used in a source-to-source transformation. In that case, we should actually lift the `trackOrigin` parameter, and set it to `true` in the solvers only. Perhaps that is the safest to do anyway, to keep this code usable for other transformations in the future.

Reply:
I've lifted the `trackOrigin` parameter, so that the solver can call the `apply()` methods of the ApplyModes with `trackOrigin` set to true.